org.mwc.debrief.core/src/org/mwc/debrief/core/editors/PlotEditor.java

```diff
@@ -1505,50 +1505,17 @@ else if (input instanceof FileStoreEditorInput)
 
 								final IFile file = ((IFileEditorInput) getEditorInput())
 										.getFile();
-
-								// get the current path (since we're going to be moving the temp
-								// one to it
-								final IPath thePath = file.getLocation();
-
-								// create a backup path
-								final IPath bakPath = file.getFullPath()
-										.addFileExtension("bak");
-
-								// delete any existing backup file
-								final File existingBackupFile = new File(file.getLocation()
-										.addFileExtension("bak").toOSString());
-								if (existingBackupFile.exists())
-								{
-									CorePlugin.logError(Status.INFO,
-											"Existing back file still there, having to delete"
-													+ existingBackupFile.getAbsolutePath(), null);
-									existingBackupFile.delete();
-								}
-
-								// now rename the existing file as the backup
-								file.move(bakPath, true, monitor);
-
-								// move the temp file to be our real working file
-								final File destFile = thePath.toFile().getAbsoluteFile();
-								if (!tmpFile.renameTo(destFile))
-								{
-									FileUtils.moveFile(tmpFile, destFile);
-								}
-
-								// finally, delete the backup file
-								if (existingBackupFile.exists())
-								{
-									CorePlugin.logError(Status.INFO,
-											"Save operation completed successfully, deleting backup file"
-													+ existingBackupFile.getAbsolutePath(), null);
-									existingBackupFile.delete();
+
+								InputStream source = null;
+								try
+								{
+									source = new FileInputStream(tmpFile);
+									file.setContents(source, true, true, monitor);
+								}
+								finally
+								{
+									FileUtil.safeClose(source);

```

Comment:
This just close a stream without an exception, what we want here. I use such code in a lot places (in Debrief or other app).
The warning is the standard Eclipse warning when using internal Eclipse API. If we won't use the internal API we can replace this code with original code. However we use FileUtil.transferStreams (line 1540) that is slightly more complex. 
So, the code is correct, but question is if you want me to remove internal API. I think that we use it in other places too, so it isn't important here.